src/components/autocomplete/js/autocompleteController.js

```diff
@@ -948,11 +951,37 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
    */
   function updateScroll () {
     if (!elements.li[0]) return;
+    if (mode === MODE_STANDARD) {
+      updateStandardScroll();
+    } else {
+      updateVirtualScroll();
+    }
+  }
+
+  function updateVirtualScroll() {
+    // elements in virtual scroll have consistent heights
     var height = elements.li[0].offsetHeight,
         top = height * ctrl.index,
         bot = top + height,
         hgt = elements.scroller.clientHeight,
         scrollTop = elements.scroller.scrollTop;
+
+    if (top < scrollTop) {
+      scrollTo(top);
+    } else if (bot > scrollTop + hgt) {
+      scrollTo(bot - hgt);
+    }
+  }
+
+  function updateStandardScroll() {
+    // elements in standard scroll have variable heights
+    var selected =  elements.li[ctrl.index] || elements.li[0];
+    var height = elements.scrollContainer.offsetHeight,
+        top = selected && selected.offsetTop || 0,
+        bot = top + selected.clientHeight,
+        hgt = height,
```

Comment:
Can you please rename `bot` in both update scroll functions? `bottom` should be more clear.

I think that `hgt` can be removed here and `height` used below?

Reply:
No problem, I realize that most of that naming was there already.